azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/AbstractFunctionMojo.java

```diff
@@ -228,17 +253,23 @@ protected File getArtifact() throws AzureToolkitRuntimeException {
         if (artifact.getFile() != null) {
             return artifact.getFile();
         }
-        // Get artifact by buildDirectory and finalName
-        // as project.getArtifact() will be null when invoke azure-functions:package directly
-        final String finalName = project.getBuild().getFinalName();
-        final String packaging = project.getPackaging();
-        final File result = new File(buildDirectory, StringUtils.join(finalName, FilenameUtils.EXTENSION_SEPARATOR, packaging));
+        String finalNameWithExtension = finalNameWithExtension();
+        final File result = new File(buildDirectory, finalNameWithExtension);
         if (!result.exists()) {
             throw new AzureToolkitRuntimeException(CAN_NOT_FIND_ARTIFACT);
         }
         return result;
     }
 
+    protected String finalNameWithExtension() {
```

Comment:
May be we could add a parameter like `nativeExecutablePath' instead of boolean `nativeExecutable`, as we could only guess the final artifact here, but users may modify it with configuration in quarkus maven plugin

Reply:
So, just to validate if I undestand all:
1. Move getArtifact from AbstractFunctionMojo to Package
2. change nativeExecutable to nativeExecutablePath, and if nativeExecutablePath is defined, I will call copyBinaryToStageDirectory instead copyJarsToStageDirectory

Correct?

Reply:
1. I prefer to keep `getArtifact` as it is, and move binary related logic to `copyBinaryToStageDirectory` in package mojo.
2. Yes, I think `nativeExecutablePath` may be more flexible, and we no longer need to generate the path for executable

Reply:
Hi @Flanker32 , I did not test all the changes yet.